agent/agent.go

```diff
@@ -440,6 +440,7 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, initialTC *taskCon
 	}
```

Comment:
I'm leaving this as a comment so it is collapsable (as with the others)

This is a task that succeeded (and it had a post command):

```
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-28T22:56:37.301Z",
  "scheduled_time": "2024-03-28T22:55:54.88Z",
  "container_allocated_time": null,
  "start_time": "2024-03-28T22:56:37.69Z",
  "finish_time": "2024-03-28T22:56:40.768Z",
  "ingest_time": "2024-03-28T22:21:01.818Z",
  "activated_time": "2024-03-28T22:55:52.656Z",
  "version_id": "6605ed477e708300076689cd",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01",
  "distro_id": "ubuntu2204-small",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0b1a77c0db5166034",
  "pod_id": "",
  "execution": 1,
  "order": 142,
  "status": "success",
  "display_status": "success",
  "status_details": {
    "status": "success",
    "type": "",
    "desc": "",
    "error_block_type": "",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "ad8a29ede4d6667b7cb56ed823b71b41",
    "disk_devices": [
      "xvda1",
      "xvdb"
    ]
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/system"
  },
  "time_taken_ms": 3078,
  "expected_duration_ms": 12927,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```

Reply:
A task that failed in the post: 

```
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-28T22:56:40.939Z",
  "scheduled_time": "2024-03-28T22:55:54.88Z",
  "container_allocated_time": null,
  "start_time": "2024-03-28T22:56:41.569Z",
  "finish_time": "2024-03-28T22:56:44.887Z",
  "ingest_time": "2024-03-28T22:21:45.025Z",
  "activated_time": "2024-03-28T22:55:50.016Z",
  "version_id": "6605ed727e708300076689d4",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45",
  "distro_id": "ubuntu2204-small",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0b1a77c0db5166034",
  "pod_id": "",
  "execution": 2,
  "order": 143,
  "status": "failed",
  "display_status": "failed",
  "status_details": {
    "status": "failed",
    "type": "test",
    "desc": "'shell.exec' (step 1 of 1) in block 'post'",
    "error_block_type": "post",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "ecbfde5ad54ea642e08c1a878e61ce0d",
    "disk_devices": [
      "xvda1",
      "xvdb"
    ]
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/system"
  },
  "time_taken_ms": 3318,
  "expected_duration_ms": 12927,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```

Reply:
A task that failed in the pre
```
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-28T22:56:45.066Z",
  "scheduled_time": "2024-03-28T22:55:50.652Z",
  "container_allocated_time": null,
  "start_time": "2024-03-28T22:56:45.604Z",
  "finish_time": "2024-03-28T22:56:45.924Z",
  "ingest_time": "2024-03-22T18:56:17.518Z",
  "activated_time": "2024-03-28T22:55:48.225Z",
  "version_id": "65fdd44ab7233f0007285364",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17",
  "distro_id": "ubuntu2204-small",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0b1a77c0db5166034",
  "pod_id": "",
  "execution": 2,
  "order": 138,
  "status": "failed",
  "display_status": "setup-failed",
  "status_details": {
    "status": "failed",
    "type": "setup",
    "desc": "this should be set",
    "error_block_type": "pre",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "0cb27240f49a758c80b29e0cc9a354fb",
    "disk_devices": [
      "xvda1",
      "xvdb"
    ]
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/system"
  },
  "time_taken_ms": 320,
  "expected_duration_ms": 12000,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```